pkg/operator/controller/callbacks.go

```diff
@@ -132,6 +107,44 @@ func reconcileDeleteControllerDeployment(args *callbacks.ReconcileCallbackArgs)
 	return nil
 }
 
+func deleteCRDs(args *callbacks.ReconcileCallbackArgs) error {
+	crdNames := []string{
+		"dataimportcrons.cdi.kubevirt.io",
+		"datavolumes.cdi.kubevirt.io",
+		"objecttransfers.cdi.kubevirt.io",
+	}
+	crdsExist := false
+	for _, crdName := range crdNames {
+		crd := &extv1.CustomResourceDefinition{}
+		if err := args.Client.Get(context.TODO(), types.NamespacedName{Name: crdName}, crd); err != nil {
+			if errors.IsNotFound(err) {
+				continue
+			}
+			return err
+		}
+		if crd.DeletionTimestamp == nil {
+			if err := args.Client.Delete(context.TODO(), crd, &client.DeleteOptions{}); err != nil {
+				if errors.IsNotFound(err) {
+					continue
+				}
+				return err
+			}
+			if err := args.Client.Get(context.TODO(), types.NamespacedName{Name: crdName}, crd); err != nil {
+				if errors.IsNotFound(err) {
+					continue
+				}
+				return err
+			}
+			args.Logger.Info("CRD is not deleted yet", "crdName", crdName)
+			crdsExist = true
+		}
```

Comment:
you have to set `crdsExist = true` here is else block when `crd.DeletionTimestamp != nil`

Reply:
Fat fingers. Fixed.